api/src/controllers/team.js

```diff
@@ -59,4 +61,28 @@ async function getTeam(req, res) {
 
 }
 
-module.exports = { storeHandle, getTeam };
\ No newline at end of file
+/**
+ * Handles a PATCH request to update a team's details on the endpoint /team/:id.
+ * @param {*} req request object - team id in url, json object containing atleast one field from team model.
+ * @param {*} res response object - updated team object
+ * @returns 200: returns updated team details
+ * @returns 404: team not found
+ * @returns 400: error updating team
+ */
+async function updateTeam(req, res) {
+    const { id: _id } = req.params;
+    const updates = req.body;
+    const options = {new: true};
+
+    if (!mongoose.Types.ObjectId.isValid(_id))
+    return res.status(404).send("Error: No team with that id.");
+
+    try {
+        const result = await Team.findByIdAndUpdate(_id, updates, options);
```

Comment:
Is `findByIdAndUpdate` our method or some in-built one?

Reply:
It's an inbuilt method from Mongoose.